File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 20 and 21 April 2019/05. Easter Bake/05. Easter Bake.js
function easterBake(arg) {

    let cakeNumbers = arg.shift()
    let flourPacket = 950
    let sugarPacket = 750
    let neededSugarPkg = 0
    let neededFlourPkg = 0
    let maxFlour = []
    let maxSugar = []

    for (let i = 0; i < cakeNumbers; i++) {
        maxSugar.push(+arg.shift())
        maxFlour.push(+arg.shift())
    }

    neededFlourPkg = Math.ceil(maxFlour.reduce((a, e) => a + e) / sugarPacket)
    neededSugarPkg = Math.ceil(maxSugar.reduce((a, e) => a + e) / flourPacket)

    console.log(`Sugar: ${neededSugarPkg}`)
    console.log(`Flour: ${neededFlourPkg}`)
    console.log(`Max used flour is ${Math.max(...maxFlour)} grams, max used sugar is ${Math.max(...maxSugar)} grams.`)
}

//easterBake([3, 400, 350, 250, 300, 450, 380])

File: Python Exams/01. Basic Exam Collection/Basics Online Exam - 20 and 21 April 2019/05. Easter Bake/05. Easter Bake.py
from math import ceil

cake_numbers = int(input())
flour_packet = 950
sugar_paket = 750
needed_sugar_pkg = 0
needed_flour_pkg = 0
max_flour = []
max_sugar = []

for _ in range(cake_numbers):
    max_sugar.append(int(input()))
    max_flour.append(int(input()))


needed_flour_pkg = ceil(sum(max_flour) / sugar_paket)
needed_sugar_pkg = ceil(sum(max_sugar) / flour_packet)

print(f'Sugar: {needed_sugar_pkg}')
print(f'Flour: {needed_flour_pkg}')
print(f'Max used flour is {max(max_flour)} grams, max used sugar is {max(max_sugar)} grams.')



'''
3
400
350
250
300
450
380
'''
